pkg/porter/mixins.go

```diff
@@ -106,3 +117,93 @@ func (p *Porter) GenerateMixinFeed(opts feed.GenerateOptions) error {
 func (p *Porter) CreateMixinFeedTemplate() error {
 	return feed.CreateTemplate(p.Context)
 }
+
+// MixinsCreateOptions represent options for Porter's mixin create command
+type MixinsCreateOptions struct {
+	MixinName  string
+	AuthorName string
+	DirPath    string
+}
+
+func (o *MixinsCreateOptions) Validate(args []string, cxt *context.Context) error {
+	if len(args) < 1 || args[0] == "" {
+		return errors.New("mixin name is required")
+	}
+
+	if len(args) > 1 {
+		return errors.Errorf("only one positional argument may be specified, the mixin name, but multiple were received: %s", args)
+	}
+
+	o.MixinName = args[0]
+
+	if o.AuthorName == "" {
+		return errors.New("must provide a value for --author")
+	}
+
+	if o.DirPath == "" {
+		o.DirPath = cxt.Getwd()
+	}
+
+	if _, err := cxt.FileSystem.Stat(o.DirPath); err != nil {
+		return errors.Wrapf(err, "invalid --dir: %s", o.DirPath)
+	}
+
+	return nil
+}
+
+func (p *Porter) CreateMixin(opts MixinsCreateOptions) error {
+	skeletorDestPath := opts.DirPath + "/" + opts.MixinName
+
+	if err := exec.Command("git", "clone", SkeletorRepo, skeletorDestPath).Run(); err != nil {
+		return errors.Wrapf(err, "failed cloning skeletor repo")
+	}
+
+	err := os.Rename(skeletorDestPath+"/cmd/skeletor", skeletorDestPath+"/cmd/"+opts.MixinName)
+	if err != nil {
+		return err
+	}
+
+	err = os.Rename(skeletorDestPath+"/pkg/skeletor", skeletorDestPath+"/pkg/"+opts.MixinName)
+	if err != nil {
+		return err
+	}
+
+	replacementList := map[string]string{
+		"get.porter.sh/mixin/skeletor":       fmt.Sprintf("github.com/%s/%s", opts.AuthorName, opts.MixinName),
```

Comment:
We will need one more flag to collect the user's github username. --author has their full name, e.g. "Carolyn Van Slyck", which won't work as a replacement for the package name. How about `--username`?